client/blocks/domain-to-plan-nudge/index.jsx

```diff
@@ -0,0 +1,160 @@
+import { submitTransaction } from 'lib/upgrades/actions/checkout';
+import PlanPrice from 'my-sites/plan-price';
+import PlanIcon from 'components/plans/plan-icon';
+
+const debug = debugFactory( 'calypso:domain-to-plan-nudge' );
+
+class DomainToPlanNudge extends Component {
+
+	constructor() {
+		super( ...arguments );
+		this.oneClickUpgrade = this.oneClickUpgrade.bind( this );
+		this.handleTransactionComplete = this.handleTransactionComplete.bind( this );
+	}
+
+	static propTypes = {
+		hasFreePlan: PropTypes.bool,
+		site: PropTypes.object,
+		siteId: PropTypes.number,
+		translate: PropTypes.func
+	};
+
+	isVisible() {
+		const { site, hasFreePlan } = this.props;
+		return site &&			//site exists
+			site.wpcom_url &&   //has a mapped domain
+			hasFreePlan;        //has a free wpcom plan
+	}
+
+	handleTransactionComplete( error ) {
+		//TODO: match expected analytics calls
+		debug( 'transaction complete', error );
+	}
+
+	oneClickUpgrade() {
+		const { siteId, storedCard } = this.props;
+		const newPayment = storedCardPayment( storedCard );
+		const transaction = {
+			payment: newPayment
+		};
+
+		// set this to temporary so we don't clear the existing user cart on the server
+		let cart = emptyCart( siteId, { temporary: true } );
+
+		const personalCartItem = {
+			free_trial: false,
+			is_domain_registration: false,
+			product_id: plansList[ PLAN_PERSONAL ].getProductId(),
+			product_slug: PLAN_PERSONAL
+		};
+		cart = add( personalCartItem )( cart );
+
+		debug( 'purchasing with', cart, transaction );
+
+		submitTransaction( { cart, transaction }, this.handleTransactionComplete );
+	}
+
+	render() {
+		if ( ! this.isVisible() ) {
+			return null;
+		}
+
+		const { translate, storedCard } = this.props;
+		return (
+			<Card className="domain-to-plan-nudge">
+				<QueryStoredCards />
+
+				<div className="domain-to-plan-nudge__header">
+					<div className="domain-to-plan-nudge__header-icon">
+						<PlanIcon plan="personal-bundle" />
+					</div>
+					<div className="domain-to-plan-nudge__header-copy">
+						<h3 className="domain-to-plan-nudge__header-title">
+							{ translate( 'Upgrade to a Personal Plan and Save!' ) }
+						</h3>
+						<ul className="domain-to-plan-nudge__header-features">
+							<li>
+								{ translate( 'Remove all WordPress.com advertising from your website' ) }
+							</li>
+							<li>
+								{ translate( 'Get high quality live chat and priority email support' ) }
+							</li>
+							<li>
+								{ translate( 'Upload up to 3GB of photos and videos' ) }
+							</li>
+							<li>
+								{ translate( 'Bundled with your domain for the best value!' ) }
+							</li>
+						</ul>
+					</div>
+				</div>
+				<div className="domain-to-plan-nudge__actions-group">
+					<div className="domain-to-plan-nudge__plan-price-group">
+						<div className="domain-to-plan-nudge__discount-percentage">
+							Save 25%
+						</div>
+						<PlanPrice rawPrice="35.88" original />
```

Comment:
Yep, I have commit ready that replaces the hard-coded string with proper rawPrice. Will do this tomorrow.